Multithreaded compile fails (make -j)

When using a large number of jobs with `make -j<N>` (e.g. `-j20`), I encounter one of two compile errors:
```
In file included from ./src/verilog/parse/parser.h:40:0,
                 from src/runtime/runtime.cc:46:
./src/verilog/parse/lexer.h:47:5: error: ‘yyParser’ does not name a type
     yyParser::symbol_type yylex(Parser* parser);
     ^
compilation terminated due to -Wfatal-errors.
Makefile:144: recipe for target 'src/runtime/runtime.o' failed
make: *** [src/runtime/runtime.o] Error 1
make: *** Waiting for unfinished jobs....
In file included from ./src/verilog/parse/parser.h:40:0,
                 from src/target/common/remote_runtime.cc:44:
./src/verilog/parse/lexer.h:47:5: error: ‘yyParser’ does not name a type
     yyParser::symbol_type yylex(Parser* parser);
     ^
compilation terminated due to -Wfatal-errors.
Makefile:144: recipe for target 'src/target/common/remote_runtime.o' failed
```
Or (occasionally):
```
In file included from ./src/verilog/parse/lexer.h:38:0,
                 from ./src/verilog/parse/parser.h:40,
                 from src/target/common/remote_runtime.cc:44:
./src/verilog/parse/verilog.tab.hh:40:0: error: unterminated #ifndef
 #ifndef YY_YY_VERILOG_TAB_HH_INCLUDED
 ^
compilation terminated due to -Wfatal-errors.
Makefile:144: recipe for target 'src/target/common/remote_runtime.o' failed
make: *** [src/target/common/remote_runtime.o] Error 1
make: *** Waiting for unfinished jobs....
In file included from ./src/verilog/parse/parser.h:40:0,
                 from src/runtime/runtime.cc:46:
./src/verilog/parse/lexer.h:47:5: error: ‘yyParser’ does not name a type
     yyParser::symbol_type yylex(Parser* parser);
     ^
compilation terminated due to -Wfatal-errors.
```

Running `make` again with any number of jobs produces a working executable. This suggests that there is a race condition between when the file `src/verilog/parse/parser.h` (where `yyParser` is declared) is created, and when `src/target/common/remote_runtime.o` is attempted to be built. This seems like a Makefile dependency issue.

--- a/Makefile
+++ b/Makefile
@@ -138,9 +138,9 @@ ${FLEX_SRC}: src/verilog/parse/verilog.ll ${BISON_SRC}
 	cd src/verilog/parse && flex verilog.ll	
 ${BISON_SRC}: src/verilog/parse/verilog.yy
 	cd src/verilog/parse && bison -d -v verilog.yy
-%.o: %.c 
+%.o: %.c ${FLEX_SRC} ${BISON_SRC}
 	${CC} ${CFLAGS} ${CC_OPT} ${PERF} ${GTEST_INC} ${INC} -c $< -o $@
-%.o: %.cc 
+%.o: %.cc ${FLEX_SRC} ${BISON_SRC} 
 	${CXX} ${CXXFLAGS} ${CXX_OPT} ${PERF} ${GTEST_INC} ${INC} -c $< -o $@
 ${GTEST_LIB}: submodule
 	mkdir -p ${GTEST_BUILD_DIR}

